### DISCUSSION TITLE
Possibility to select specific syntax for exporting to .sql

### DISCUSSION THREAD

**Discussion by simon-spier0** [dbeaver/dbeaver#19622] [Ideas]
Hi,

I sometimes need to export data from MySQL database to the .sql file but with MSSQL syntax. So currently I export it with the original syntax (MySQL) and I let an online tool to convert it to MSSQL syntax.

Could you please add an option to select which syntax I want to export to?

Thanks.

**Comment by arhayka** [comment]
Hi @simon-spier0,
Could you please provide an example of source SQL and the expected result?

**Comment by simon-spier0** [comment]
Of course.

This is original T-SQL to create table in MS SQL:
```
CREATE TABLE dev_bgen.dbo.auth_user (
                id int IDENTITY(1,1) NOT NULL,
                password nvarchar(128) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                last_login datetimeoffset NULL,
                is_superuser bit NOT NULL,
                username nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                first_name nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                last_name nvarchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                email nvarchar(254) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                is_staff bit NOT NULL,
                is_active bit NOT NULL,
                date_joined datetimeoffset NOT NULL,
                CONSTRAINT PK__auth_use__3213E83F9E8D84AF PRIMARY KEY (id),
                CONSTRAINT auth_user_username_6821ab7c_uniq UNIQUE (username)
);
```

I would like to change output syntax to MySQL:
```
CREATE TABLE dev_bgen.auth_user (
                id int AUTO_INCREMENT NOT NULL,
                password varchar(128) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                last_login datetime(6) NULL,
                is_superuser tinyint NOT NULL,
                username varchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                first_name varchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                last_name varchar(150) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                email varchar(254) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
                is_staff tinyint NOT NULL,
                is_active tinyint NOT NULL,
                date_joined datetime(6) NOT NULL,
                CONSTRAINT PK__auth_use__3213E83F9E8D84AF PRIMARY KEY (id),
                CONSTRAINT auth_user_username_6821ab7c_uniq UNIQUE (username)
);
```

**Comment by LonwoLonwo** [comment]
Hello @simon-spier0 

We do not have such functionality and do not plan to add it in the future. Because the variety of databases syntaxes is so huge that we cannot embrace them all. This will be a titanic work, even though, so far, it is unclear how many people are interested in this feature request.

For now, you can change your export data with settings with the Export SQL instrument:

![2023-04-06 12_31_49-DBeaver Ultimate 23 0 2 - _default 1_ Script-216  Script-216](https://user-images.githubusercontent.com/45152336/230336872-879df023-6c77-4a4c-a975-eabbab851844.png)

Also, you can configure different DDLs for your tables during the export from one database to another:

![2023-04-06 12_32_48-Configure metadata structure](https://user-images.githubusercontent.com/45152336/230337087-5f5847d5-c2f6-4420-a794-a5b4b4197b3a.png)
